### PR TITLE
Acsone addons distribution list improvements

### PR DESCRIPTION
**Pull request by JonathanNEMRY**
**Improvements of `distribution_list`**
- Rename 'Source Model' into 'Model' for `distribution_list_line`
- Found a native solution for custom CSS (hide elements into create/edit mode)
- Hide `see result filter` into edit/create mode to avoid unwnated saves for user
- Reset `domain` when model change into `distribution_list_line`
- Add a `widget selection` for `many2one` on `ir.model` (more safe)
- Fix tests with `notify_email` because of this partner selection's field now contains only two values
- Can now save an expression by a dynamic way depending the `ir.model` which is present on the `distribution_list_line`

### CODE REVIEW

File: distribution_list/distribution_list_view.xml [hunk 1]
```diff
@@ -109,7 +108,7 @@
                     <field name="src_model_id"/>
 
                     <group expand="0" string="Group By..." name="group_by">
-                        <filter name="src_model" string="Source Model" domain="[]" context="{'group_by':'src_model_id'}" />
+                        <filter name="src_model" string="Model" domain="[]" context="{'group_by':'src_model_id'}" />
```
Comment (laetitia-gangloff) [inline distribution_list/distribution_list_view.xml:111]: Why set a string if it to set the same than it is defined in column definition ?

Comment (JonathanNEMRY) [inline distribution_list/distribution_list_view.xml:111]: This is a string for a filter element so it is required :)

**Comment by laetitia-gangloff** [comment]
:+1:

File: distribution_list/tests/test_distribution_list.py [hunk 2]
```diff
@@ -250,16 +254,131 @@ def test_compute_distribution_list_ids(self):
         self.assertEqual(id_customer in list_ids_cust_nosupl,
                          True, "The ids computed must be one customer only")
 
+    def test_not_safe_mode(self):
+        """
+        ==================
+        test_not_safe_mode
+        ==================
+        test that get_ids_from_distribution_list will with safe_mode
+        is false will not exclude id of other distribution list
```
Comment (lmignon) [inline distribution_list/tests/test_distribution_list.py]: ??? will with safe_mode is Flase will ??? Do you mean: test if 'get_ids_from_distribution_list' will not exclude ids from other distribution list when the function is called with safe_mode to False?

File: distribution_list/distribution_list.py [hunk 3]
```diff
@@ -450,6 +455,20 @@ def _get_record(self, record_or_list):
         'domain': "[]",
     }
 
+    def onchange_src_model_id(self, cr, uid, ids, context=None):
+        """
+        =====================
+        onchange_src_model_id
+        =====================
```
Comment (lmignon) [inline distribution_list/distribution_list.py]: I think this kind of formatting comments is unusual... We usually avoid specifying the name of the function in its docstring.

File: distribution_list/distribution_list.py [hunk 4]
```diff
@@ -474,7 +493,21 @@ def create(self, cr, uid, vals, context=None):
...
             vals['domain'] = '[]'
         return super(distribution_list_line, self).create(
-            cr, uid, vals, context=None)
+            cr, uid, vals, context=context)
+
+    def write(self, cr, uid, ids, vals, context=None):
+        """
+        =====
+        write
+        =====
```
Comment (lmignon) [inline distribution_list/distribution_list.py]: see previous comment on line #L460-462

File: distribution_list/distribution_list.py [hunk 5]
```diff
@@ -517,28 +550,28 @@ def get_list_from_domain(self, cr, uid, ids, context=None):
...
-        res: Launch an action act_windows with special parameters:
-               * view_mode      --> tree_partner_selection
-                   View Customized With JavaScript and QWeb
-
-               * flags          --> search_view
-                   Put the search_view to true allow to show
-                   The SearchBox into a PopUp window
+        ========================
+        action_partner_selection
+        ========================
```
Comment (lmignon) [inline distribution_list/distribution_list.py]: see previous comment on line #L460-462

**Comment by lmignon** [comment]
:+1:

**Pull request merged**